pkgs/by-name/ma/maven/build-maven-package.nix

```diff
@@ -93,6 +94,7 @@ stdenv.mkDerivation (builtins.removeAttrs args [ "mvnFetchExtraArgs" ] // {
     runHook preBuild
 
     mvnDeps=$(cp -dpR ${fetchedMavenDeps}/.m2 ./ && chmod +w -R .m2 && pwd)
+    ${afterDepsSetup}
```

Comment:
where and why was this added?

Reply:
As stated above:

> I had to extend `buildMavenPackage` with `afterDepsSetup`, this is a string of build phase commands that are injected after the nvmDeps variable is set up. This is needed as dependency-track builds its own BOM during its build and expects this file to be presented in a later build step.

Reply:
Since it changes maven builder - a thing that will affect all packages that use Maven - it should be in a separate **pull request**, so that it can be bisect-able in the future.

Reply:
Why should it be a seperate PR? A seperate commit should be bisectable as well.

Reply:
Mostly because GitHub is a mess :(
GitHub is PR- and merge-centric.

Because of it, it is better to at least open a PR for the change in Maven, so that the discussions about Maven (that affects all Java+Maven packages) can happen without clobbering the discussion on the inclusion of this package.

This is more preeminent given that when I tried to run nixpkgs-review it got stuck building many unrelated packages.

(this is why so many oldschoool developers use e-mail...)

Reply:
Another thing: GitHub allows comments on commits, but only after they have been committed.